src/pcm.c

```diff
@@ -1218,6 +1218,22 @@ int pcm_start(struct pcm *pcm)
     return 0;
 }
 
+/** Drains a PCM.
+ * @param pcm A PCM handle.
+ * @return On success, zero; on failure, a negative number.
+ * @ingroup libtinyalsa-pcm
+ */
+int pcm_drain(struct pcm *pcm)
+{
+    if (!pcm_is_ready(pcm))
+        return -1;
+
+    if (pcm->ops->drain(pcm->data) < 0)
```

Comment:
Why don't you use `pcm->ops->ioctl(pcm->data, SNDRV_PCM_IOCTL_DRAIN)`, but add a new function?

Reply:
Thanks for the comment. I have updated to simplify the logic. @rohit-qc @dvdli can you help review again.